src/analysis/normed/order/basic.lean

```diff
@@ -15,35 +15,75 @@ These are mostly useful to avoid diamonds during type class inference.
 open filter set
 open_locale topological_space
 
-/-- A `normed_linear_ordered_group` is an additive group that is both a `normed_add_comm_group` and
-    a `linear_ordered_add_comm_group`. This class is necessary to avoid diamonds. -/
-class normed_linear_ordered_group (α : Type*)
-extends linear_ordered_add_comm_group α, has_norm α, metric_space α :=
-(dist_eq : ∀ x y, dist x y = norm (x - y))
+variables {α : Type*}
 
-@[priority 100] instance normed_linear_ordered_group.to_normed_add_comm_group (α : Type*)
-  [normed_linear_ordered_group α] : normed_add_comm_group α :=
-⟨normed_linear_ordered_group.dist_eq⟩
+/-- A `normed_ordered_add_group` is an additive group that is both a `normed_add_comm_group` and an
+`ordered_add_comm_group`. This class is necessary to avoid diamonds. -/
```

Comment:
Could you specify which diamonds you want to avoid here?

Reply:
This is the usual "can't state stuff with `[ordered_add_comm_group α] [normed_add_comm_group α]` because that would give two group structures on `α`". I stole this phrasing from the original docstrings.

Reply:
How about the following then:
```suggestion
`ordered_add_comm_group`. This class is necessary to avoid diamonds caused by both classes
carrying their own group structure. -/
```